torchvision/datasets/ucf101.py

```diff
@@ -71,35 +81,30 @@ def __init__(self, root, annotation_path, frames_per_clip, step_between_clips=1,
             _video_min_dimension=_video_min_dimension,
             _audio_samples=_audio_samples,
         )
-        self.video_clips_metadata = video_clips.metadata
-        self.indices = self._select_fold(video_list, annotation_path, fold, train)
-        self.video_clips = video_clips.subset(self.indices)
-        self.transform = transform
+        self.video_clips_metadata = self.video_clips.metadata
```

Comment:
This is unfortunately a BC-breaking change as other downstream projects rely on this behavior, see [ClassyVision](https://github.com/facebookresearch/ClassyVision/blob/7632f2a3f9ff8b753d350208659e97dbf90f78a7/classy_vision/dataset/classy_ucf101.py#L102) for example.

The original thinking behind this approach was that one could cache the metadata and re-use it over different dataset invocations, so that the creation time would be amortized.

It could indeed be possible to from the beginning create separate metadata for each fold, but now that it's been done as is we unfortunately to keep it for backwards-compatibility reasons.